packages/protocol/contracts/common/UsingRegistry.sol

```diff
@@ -49,6 +53,15 @@ contract UsingRegistry is Ownable {
     _;
   }
 
+  modifier onlyRegisteredContracts(bytes32[] memory identifierHashes) {
+    bool registered = false;
+    for (uint256 i = 0; i < identifierHashes.length; i++) {
+      if (registry.getAddressForOrDie(identifierHashes[i]) == msg.sender) registered = true;
+    }
+    require(registered);
```

Comment:
If you know of a better way to do this lmk, wasn't sure how much to try and "optimize" vs favoring readability

Reply:
This looks nice to me!